reader/src/main/java/org/jline/reader/PrintAboveWriter.java

```diff
@@ -0,0 +1,58 @@
+package org.jline.reader;
+
+import java.io.IOException;
+import java.io.Writer;
+
+/**
+ * Redirects a {@link Writer} to a {@link LineReader}'s {@link LineReader#printAbove(String)} method,
+ * which draws output above the current prompt / input line.
+ *
+ * <p>Example:</p>
+ * <pre>
+ *     LineReader reader = LineReaderBuilder.builder().terminal(terminal).parser(parser).build();
+ *     PrintAboveWriter printAbove = new PrintAboveWriter(reader);
+ *     printAbove.write(new char[] { 'h', 'i', '!', '\n'});
+ * </pre>
+ *
+ */
+public class PrintAboveWriter extends Writer {
```

Comment:
It may be cleaner to extend `java.io.StringWriter`

Reply:
Good point, adjusted.